lib/chain.js

```diff
@@ -31,8 +31,8 @@ module.exports = (api) => {
     }
 
     filter(fn) {
-      this._update((array, resolve) => {
-        api.metasync.filter(array, fn, result => resolve(result));
+      this._update((array, resolve, reject) => {
```

Comment:
Note that all these methods can now be greatly simplified. I supposed them all to be one-liners, but inconsistent callback style impeded it.